src/utilstrencodings.cpp

```diff
@@ -506,3 +506,25 @@ bool ParseFixedPoint(const std::string &val, int decimals, int64_t *amount_out)
     return true;
 }
 
+/// Parse the decimal string into a little-endian byte vector.
+std::optional<std::vector<uint8_t>> ParseArbitraryInt(const std::string& num_string)
+{
+    std::vector<unsigned char> result;
+    size_t start = num_string.find_first_not_of(WHITESPACE);
+    if (start == std::string::npos) start = 0;
+    size_t end = num_string.find_last_not_of(WHITESPACE);
+    if (end == std::string::npos) end = start;
+    for (char c : num_string.substr(start, end-start+1)) {
```

Comment:
```suggestion
    for (char c : num_string.substr(start, end-start)) {
```

Reply:
I think the `+1` is needed, because consider a string like `"1"`, in this case, `start` and `end` will be zero, so the substring we want starts at 0 and has a length of 0-0+1 = 1.